File: 백준/Silver/10431. 줄세우기/줄세우기.py
import sys
input = sys.stdin.readline

T = int(input())

for _ in range(T):
    lst = list(map(int, input().split()))
    sumV = 0
    sort_list = [lst[1]]
    for idx in range(2, 21):
        for st_idx, student in enumerate(sort_list):
            if lst[idx] < student:
                sumV += len(sort_list) - st_idx
                break
        sort_list.append(lst[idx]) # 학생 추가하고
        sort_list.sort() # 오름차순 정렬
    print(lst[0], sumV)

File: 백준/Silver/10431. 줄세우기/줄세우기.js
const fs = require('fs')
const input = fs.readFileSync(process.platform === "linux" ? '/dev/stdin' : './10431input.txt').toString().trim().split("\n").map((el) => el.split(" ").map(Number))

const T = input.shift(); // 첫 번째 요소 ([4]) 제거
for (let i = 0; i < Number(T); i++) {
  const tc = input[i].shift() // 테스트 케이스 번호 제거후 tc로 반환
  let sortStudents = input[i].slice(0, 1) // (배열 쪼개기)첫 번째 학생만 담긴 배열
  const checkStudents = input[i].slice(1) // (배열 쪼개기) 두 번째 학생부터 끝까지 담긴 배열

  let sumV = 0
  checkStudents.forEach(student => 
  {
    let insertIdx = -1
    for (let i = 0; i < sortStudents.length; i++) {
      if (student < sortStudents[i]) {
        sumV += sortStudents.length - i // 학생들이 이동할 걸음 수 누적하고
        insertIdx = i
        break
      }
    }
      
    if (insertIdx !== -1) {
      sortStudents.splice(insertIdx, 0, student) // i인덱스 자리에 student 집어넣기
    } else {
      const newSortStudent = [...sortStudents, student]
      sortStudents = [...newSortStudent]
    }
  }
  )
  console.log(tc, sumV)
}